### ISSUE TITLE
Separate ping from dashboard?

### ISSUE THREAD

**Issue by deitch** [containous/traefik#2094]
### Do you want to request a *feature* or report a *bug*?
Feature (or security bug, see it as you will)

* Enable `[web]` or `--web`, and anything that can reach the health check at `/ping` has full dashboard access. 
* Disable it, and no way to have a health check

Assume I have a reasonable defense-in-depth network: Internet -> public subnet (DMZ) -> private. I put a load balancer (e.g. AWS ELB) in my pubic subnet, which, in turn, has access to my private (where traefik runs). 

I want my load balancer to use the `/ping` health check to see the liveness of traefik. But if I do, then my public subnet (and ELB) has complete access to my dashboard. If I remove it, I cannot check traefik's health.

I should be able to do one of several things:

* Define a route in traefik on its usual entrypoints that is a health check, e.g. `http://:80/ping` even though `http://:80/` normally serves all of my proxied traffic
* Define a separate port for dashboard and `/ping`, so I can expose my health check port comfortably without exposing my dashboard

**Comment by ldez** [comment]
With # 1982, you can now add authentication to the API and Dashboard, but ping is excluded from this authentication.

Consequently you have an "unsecured" `ping` and a secured web provider.

**Comment by deitch** [comment]
Thanks @idez. I did see that. Two things

1. It isn't clear how I enable ping but require auth for dashboard. `--web` turns both on. 
2. Even with that it is suboptimal. I well might want to expose `/ping` to an external validator (ELB health check or even regular monitoring service) but it is not good practice to expose the dashboard, even with authentication required. 

I think ideal in my book would be the ability to set a special health check path (like `/ping`) on the entrypoint itself. That is what is exposed anyways, so it makes sense. 

Summarizing:

1. How do I enable anonymous ping but require auth to dash?
2. Can I put ping on an entirely different port to keep separate security concerns in separate domains?
3. Is there any way to route `/ping` on the entrypoint itself around to the admin port?

**Comment by ldez** [comment]
1.
```
[web]
address = ":8000"
    [web.auth.basic]
    users = ["traefik:$apr1$8EvWcwpO$RaRFXvDJk6LGQ5tM04RRd0"]
```

2. no
3. I don't know.

**Comment by deitch** [comment]
1. auth: thanks. 
2. port: thanks. I would recommend it. 
3. route: I am trying to think of a way to do it, have not thought of it. Can I make a route for `/ping`  that routes to `localhost:8080/ping`?

**Comment by deitch** [comment]
I think I got it. For route:

```toml
[backends]
  [backends.traefik]
    [backends.traefik.servers.server1]
    url = "http://localhost:8080"
    weight = 10

[frontends]
  [frontends.traefikadmin]
  backend = "traefik"
    [frontends.traefikadmin.routes.ping]
    rule = "Path:/ping"
```

**Comment by deitch** [comment]
In any case, I am leaving this open as a "request for feature".

**Comment by deitch** [comment]
Hmm, I think this might work:

```toml
defaultEntryPoints = ["http"]

[entryPoints]
  [entryPoints.http]
  address = ":80"
  [entryPoints.ping]
  address = ":8082"

[backends]
  [backends.traefik]
    [backends.traefik.servers.server1]
    url = "http://localhost:8080"
    weight = 10

[frontends]
  [frontends.traefikadmin]
  backend = "traefik"
  entrypoints = ["ping"]
    [frontends.traefikadmin.routes.ping]
    rule = "Path:/ping"
```

**Comment by deitch** [comment]
It does. Rather nicely.

I would be happy to open a PR for these 2 patterns. Where should it do?

**Comment by ldez** [comment]
Yes you can add this examples in `user-guide/examples.md`.
Could you make your PR on the 1.4 branch?

**Comment by deitch** [comment]
Sure

**Comment by traefiker** [comment]
Closed by #2102.
